client.go

```diff
@@ -52,11 +58,82 @@ func (d *DefaultClient) GetUserData() interface{} {
 	return d.UserData
 }
 
-// Implement the ClientSecretMatcher interface
+// ClientSecretMatches implement the ClientSecretMatcher interface
 func (d *DefaultClient) ClientSecretMatches(secret string) bool {
 	return d.Secret == secret
 }
 
+// ComboClient implements osin.Client interface
+// This type of client is intended to handle multiple audience
+// in the token
+type ComboClient struct {
+	Audience jwt.Audience
+	Clients  []Client
+}
+
+// GetID satisfies osin.Client interface.
+func (client *ComboClient) GetID() string {
+	return strings.Join(client.Audience, ",")
+}
+
+// GetSecret satisfies osin.Client interface
+func (client *ComboClient) GetSecret() string {
+	secrets := make([]string, 0, len(client.Clients))
+	for _, c := range client.Clients {
+		secrets = append(secrets, c.GetSecret())
+	}
+	return strings.Join(secrets, ",")
+}
+
+// GetRedirectURI satisfies osin.Client interface
+func (client *ComboClient) GetRedirectURI() string {
+	uris := make([]string, 0, len(client.Clients))
+	for _, c := range client.Clients {
+		uris = append(uris, c.GetRedirectURI())
+	}
+	return strings.Join(uris, ",")
+}
+
+// GetUserData satisfies osin.Client interface
+func (client *ComboClient) GetUserData() interface{} {
+	data := make([]interface{}, 0, len(client.Clients))
+	for _, c := range client.Clients {
+		data = append(data, c.GetUserData())
+	}
+	return data
+}
+
+// ClientSecretMatches satisfies the ClientSecretMatcher interface
+func (client *ComboClient) ClientSecretMatches(secret string) bool {
+	for _, c := range client.Clients {
+		if verifyPassword(c.GetSecret(), secret) {
```

Comment:
why do you need to use bcrypt for matching secrets in combo client? while in the default client we can just match plain text?

Reply:
we dont actually need it to be bcrypt, it could be anything, 
for simplicity i'll implement plain text instead. thanks.